File: ProyectosJAVA/MAVEN/SoftProgBusiness/src/main/java/pe/edu/pucp/softprog/rrhh/bo/AreaBOImpl.java
package pe.edu.pucp.softprog.rrhh.bo;


import java.util.ArrayList;
import pe.edu.pucp.softprog.rrhh.dao.AreaDAO;
import pe.edu.pucp.softprog.rrhh.model.Area;
import pe.edu.pucp.softprog.rrhh.mysql.AreaImpl;
import pe.edu.pucp.softprog.rrhh.boi.IAreaBO;

public class AreaBOImpl implements IAreaBO{
    
    private final AreaDAO daoArea;
    
    public AreaBOImpl(){
        daoArea = new AreaImpl();
    }
    
    @Override
    public int insertar(Area area) throws Exception{
        validar(area);
        return daoArea.insertar(area);
    }
    
    @Override
    public int modificar(Area area) throws Exception{
        validar(area);
        return daoArea.modificar(area);
    }
    
    @Override
    public Area obtenerPorId(int idArea) throws Exception{
        return daoArea.obtenerPorId(idArea);
    }
    
    @Override
    public ArrayList<Area> listarTodos() throws Exception{
        return daoArea.listarTodos();
    }
    
    @Override
    public void validar(Area area) throws Exception{
        if(area.getNombre().trim().isEmpty())
            throw new RuntimeException("El nombre del area a registrar esta vacio.");
        if(area.getNombre().trim().length()>75)
            throw new RuntimeException("El nombre a registrar excede los 75 caracteres.");
    }

    @Override
    public int eliminar(int idArea) throws Exception {
        return daoArea.eliminar(idArea);
    }
 
}

File: ProyectosCSharp/SoftProg/SoftProgBusiness/RRHH/BOI/AreaBOImpl.cs
using SoftProgBusiness.RRHH.BO;
using SoftProgModel.RRHH;
using SoftProgPersistance.RRHH.DAO;
using SoftProgPersistance.RRHH.Impl;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgBusiness.RRHH.BOI
{
    public class AreaBOImpl : IAreaBO
    {
        private AreaDAO daoArea;
        public AreaBOImpl()
        {
            daoArea = new AreaImpl();
        }

        public int eliminar(int idArea)
        {
            return daoArea.eliminar(idArea);
        }

        public int insertar(Area area)
        {
            validar(area);
            return daoArea.insertar(area);
        }

        public BindingList<Area> listarTodos()
        {
            return daoArea.listarTodos();
        }

        public int modificar(Area area)
        {
            validar(area);
            return daoArea.modificar(area);
        }

        public Area obtenerPorId(int idObjeto)
        {
            throw new NotImplementedException();
        }

        public void validar(Area area)
        {
            if (area.Nombre.Trim().Length == 0)
                throw new Exception("El nombre del area a registrar esta vacio.");
            if (area.Nombre.Trim().Length > 75)
                throw new Exception("El nombre a registrar excede los 75 caracteres.");
        }
    }
}
